### PR TITLE
Multiple improvements of the pathing and mouse following

### PR DESCRIPTION
**Pull request by RogerNogue**
To test this feature:
Personal recomendation: delete the objects that generate particles (in order to improve performance)
- load the graveyard scene
- press play
- see that now player follows mouse even when mouse passes through the position of the player
- see that if clicking outside the map. If not too far, the player goes to the edge of the map.
- see that Dash also does that

### CODE REVIEW

File: Source/ResourceScene.cpp [hunk 1]
```diff
@@ -152,6 +152,10 @@ bool ResourceScene::Load()
 	RELEASE(json);
 
 	App->scene->root->OnPlay();
+	//set all the game objects
+	App->scene->root->UpdateTransforms(math::float4x4::identity);
```
Comment (AlbertVVila) [inline Source/ResourceScene.cpp]: This is from another PR 😢

File: Source/ModuleScene.cpp [hunk 2]
```diff
@@ -107,8 +107,10 @@ bool ModuleScene::Start()
 	if (defaultSceneUID > 0u)
 	{
 		defaultScene = (ResourceScene*)App->resManager->Get(defaultSceneUID);
-		if(defaultScene != nullptr)
+		if (defaultScene != nullptr)
+		{
```
Comment (AlbertVVila) [inline Source/ModuleScene.cpp]: this doesn't belong here 😢

Comment (RogerNogue) [reply Source/ModuleScene.cpp]: mergeing with develop it disappeared

**Comment by EnriquePuyol** [comment]
Did you make any changes to the scripts? I cant compile them because some variables have been deleted

**Comment by RogerNogue** [comment]
> Did you make any changes to the scripts? I cant compile them because some variables have been deleted

you are right, sorry. After mergeing with develop this error appeared and I did not notice

**Comment by RogerNogue** [comment]
> Dash is not longer working properly. Everything else seems fine

After mergeing with develop it is solved

**Comment by RogerNogue** [comment]
> * Recast mesh no longer exists, it would be nice to recreate it again or recover it from previous commits.
> * I think it would be interesting to solve this bug: when you hold right click on doing a Dash, you keep in the idle state but you are moving at the same time.

Yes, first point was being solved in another PR, it got accepted not long ago and merged into this branch. It's working now.

Second point; looks like its something about the animations and its start, tomorrow i'm checking it out

**Comment by RogerNogue** [comment]
walking after dash working properly now

**Pull request merged**